### PR TITLE
feat: Add more layout examples

### PR DESCRIPTION
**Pull request by RayRedGoose**
## Summary

Resolves: #2104  

This PR will add new Layout examples based on FIgma examples.

## Release Category
Components

### CODE REVIEW

File: modules/react/_examples/stories/examples/layout/CustomColumnWidth.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <BodyText size="small" textAlign="center" color="frenchVanilla100">
+          4 column width
+        </BodyText>
+      </Box>
+      <Box height="120px" backgroundColor="blueberry400">
+        <BodyText size="small" textAlign="center" color="frenchVanilla100">
+          4 column width
+        </BodyText>
+      </Box>
+      <Box height="120px" backgroundColor="blueberry500">
```
Comment (jamesfan) [inline modules/react/_examples/stories/examples/layout/CustomColumnWidth.tsx:20]: Something looks off for the `Custom Column Width` example in the preview build:

image

I think there's an extra `<Box>` which shouldn't be here:

```suggestion
      </Box>
      <Box height="120px" backgroundColor="blueberry500">
```

File: modules/react/_examples/stories/Layouts.stories.mdx [hunk 2]
```diff
@@ -26,6 +30,26 @@ Below are layout examples to reference as you build your own. For more in-depth
...
 
+## Full Width With 3 Columns
+
+<ExampleCodeBlock code={FullWidthWith3Columns} />
+
+## Full Width With 3 Columns and 2 Rows
+
+<ExampleCodeBlock code={FullWidthWith3Columns2Rows} />
+
+## Mansory
```
Comment (jamesfan) [inline modules/react/_examples/stories/Layouts.stories.mdx:41]:
```suggestion
## Masonry
```

File: modules/react/_examples/stories/Layouts.stories.mdx [hunk 3]
```diff
@@ -26,6 +30,26 @@ Below are layout examples to reference as you build your own. For more in-depth
...
+
+<ExampleCodeBlock code={Masonry} />
+
 ## Responsive Columns
 
 <ExampleCodeBlock code={ResponsiveColumns} />
+
+## Tiled 2 and 3 Columns
```
Comment (jamesfan) [inline modules/react/_examples/stories/Layouts.stories.mdx:49]: Nit: for consistency with `Heading` text in the corresponding example.
```suggestion
## 3 and 2 Column Tiled View
```

File: modules/react/_examples/stories/examples/layout/CustomColumnWidth.tsx (hunk 1 above)
Comment (RayRedGoose) [reply modules/react/_examples/stories/examples/layout/CustomColumnWidth.tsx]: yeah, it was box duplication there. Thanks for catching it

File: modules/react/_examples/stories/examples/layout/Tiled4and2Columns.tsx
[the remark this answers is not in the document]
Comment (RayRedGoose) [reply modules/react/_examples/stories/examples/layout/Tiled4and2Columns.tsx:1]: Fixed it. It was wrong column width and incorrect headings 👍

**Review by jamesfan — APPROVED** [review]
Looks good! 👍

**Comment by mannycarrera4** [comment]
lets add reference to these examples in grid.mdx  since they're mentioned in our upgrade guides

**Pull request merged**